app/preference.go

```diff
@@ -10,12 +10,11 @@ import (
 )
 
 func (a *App) GetPreferencesForUser(userId string) (model.Preferences, *model.AppError) {
-	result := <-a.Srv.Store.Preference().GetAll(userId)
-	if result.Err != nil {
-		result.Err.StatusCode = http.StatusBadRequest
-		return nil, result.Err
+	if result, err := a.Srv.Store.Preference().GetAll(userId); err != nil {
```

Comment:
nit: now is clear that the result object is a "preferences" object, so we can use the 'preferences' variable name.